File: AMap-Android-Kit/app/src/main/java/kain/amap_android_kit/MainActivity.java
package kain.amap_android_kit;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //显示地图功能
        Button viewMap = (Button)findViewById(R.id.viewMap);
        viewMap.setOnClickListener(new ButtonOnClickListener());
        //显示蓝点
        Button btShowMyLocation = (Button)findViewById(R.id.bt_showMyLocation);
        btShowMyLocation.setOnClickListener(new ButtonOnClickListener());
        //bt_changeLayer 切换地图图层
        Button btChangeLayer = (Button)findViewById(R.id.bt_changeLayer);
        btChangeLayer.setOnClickListener(new ButtonOnClickListener());
    }

    /**
     * 设置多个按钮监听事件
     */
    private class  ButtonOnClickListener implements View.OnClickListener{

        @Override
        public void onClick(View view) {
            Intent intent;
            switch (view.getId()) {
                case R.id.viewMap:
                    intent = new Intent(MainActivity.this, BaseMapActivity.class);
                    startActivity(intent);
                    break;

                case R.id.bt_showMyLocation:
                    intent = new Intent(MainActivity.this, ShowMyLocationActivity.class);
                    startActivity(intent);
                    break;

                case R.id.bt_changeLayer:
                    intent = new Intent(MainActivity.this, ChangeLayerActivity.class);
                    startActivity(intent);
                    break;

            }
        }
    }
}

File: AMap-Android-Kit-Kotlin/app/src/main/java/kain/amap_android_kit_kotlin/MainActivity.kt
package kain.amap_android_kit_kotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.os.PersistableBundle
import android.util.Log
import com.amap.api.maps.MapView

class MainActivity : AppCompatActivity() {
    private var mapView: MapView? = null
    private var mapOptions: MapOptions? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        mapView = findViewById(R.id.map) as MapView
        mapView?.onCreate(savedInstanceState)
        mapOptions = MapOptions(mapView?.map!!)
        mapOptions?.showPositionDot()
        mapOptions?.asyncMapLocation?.subscribe {
            Log.wtf("Main Msg", it.latitude.toString() + ":" + it.longitude.toString())
        }
        //mapOptions?.setGestureScaleByMapCenter(100,100,true)
        //改变地图的缩放级别
        //mapOptions?.mapZoomTo(17F)
        //设置地图显示区域
        //mapOptions?.setMapStatusLimits(33.789925, 104.838326,38.740688, 114.647472)
        //设置默认显示地点
        mapView = mapOptions?.setDefaultMap(this, 18.312963, 109.616185, 12f)
        mapView?.onCreate(savedInstanceState)


    }

    override fun onDestroy() {
        super.onDestroy()
        mapView?.onDestroy()
    }

    override fun onResume() {
        super.onResume()
        mapView?.onResume()
    }

    override fun onPause() {
        super.onPause()
        mapView?.onPause()
    }

    override fun onSaveInstanceState(outState: Bundle?, outPersistentState: PersistableBundle?) {
        super.onSaveInstanceState(outState, outPersistentState)
        mapView?.onSaveInstanceState(outState)
    }
}
